pkg/raw/raw_exec.go

```diff
@@ -5,44 +5,98 @@ package raw
 )
 
-type Exec struct {
-}
+type Exec struct{}
+
+const filePerm = 0660
 
 func (Exec) Create(filename string, opts ...CreateOption) error {
 	o := &CreateOptions{}
 	o.ApplyOptions(opts)
-	var seek string
-
-	if o.SourceFile == "" && o.Size == nil {
-		return fmt.Errorf("must specify Size when creating without source file")
-	}
+	log := ctrl.Log.WithName("raw-disk")
 
-	ofilename := "of=" + filename
-	if o.Size != nil {
-		seek = "seek=" + strconv.FormatInt(*o.Size, 10) // TODO: verify if the disk size is proper, else G as a suffix
-	}
 	if o.SourceFile == "" {
-		outMsgDD, err := exec.Command("dd", "if=/dev/zero", ofilename, "bs=1", "count=0", seek).Output()
+		if o.Size == nil {
+			return fmt.Errorf("must specify Size when creating without source file")
+		}
+		seek := *o.Size
+		// Position the file cursor one byte before the desired seek position to write a single byte,
+		// to ensure that data is written at the exact byte position specified by seek.
+		err := createEmptyFileWithSeek(log, filename, seek-1)
 		if err != nil {
-			return fmt.Errorf("failed creating the ephemeral disk: %s, exit error %w", string(outMsgDD), err)
+			return fmt.Errorf("failed creating the empty ephemeral disk at %s: %w", filename, err)
 		}
 	} else {
-		ifilename := "if=" + o.SourceFile
-		outMsgDD, err := exec.Command("dd", ifilename, ofilename).Output()
+		err := copyFile(log, o.SourceFile, filename)
 		if err != nil {
-			return fmt.Errorf("failed creating the image for virtual disk: %s, exit error %w", string(outMsgDD), err)
+			return fmt.Errorf("failed creating virtual disk image, source: %s, destination: %s: %w", o.SourceFile, filename, err)
 		}
 	}
 
 	return nil
 }
 
+func createEmptyFileWithSeek(log logr.Logger, filename string, seek int64) error {
+	log = log.WithName("createEmptyFileWithSeek").WithValues("filename", filename)
+	dstFile, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, filePerm)
+	if err != nil {
+		return fmt.Errorf("failed opening destination file: %w", err)
+	}
+
+	defer func() {
+		destErr := dstFile.Close()
+		if destErr != nil {
+			log.Error(destErr, "error closing file")
+		}
+	}()
+
+	_, err = dstFile.Seek(seek, io.SeekStart)
+	if err != nil {
+		return fmt.Errorf("failed seeking destination file: %w", err)
+	}
+
+	_, err = dstFile.Write([]byte{0})
+
+	return err
+}
+
+func copyFile(log logr.Logger, src, dst string) error {
+	log = log.WithName("copyFile").WithValues("source", src, "destination", dst)
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return fmt.Errorf("failed opening source file: %w", err)
+	}
+	defer func() {
+		srcErr := srcFile.Close()
+		if srcErr != nil {
+			log.Error(srcErr, "error closing source file")
+		}
+	}()
+
+	dstFile, err := os.OpenFile(dst, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, filePerm)
+	if err != nil {
+		return fmt.Errorf("failed opening destination file: %w", err)
+	}
+
+	defer func() {
+		destErr := dstFile.Close()
+		if destErr != nil {
```

Comment:
```suggestion
		if err := dstFile.Close(); err != nil {
```

Reply:
done

Reply:
don't use shadowing you can very easy did mistake in code because one stupid character : can completely change logic.

Reply:
Yes, I agree and had implemented this way only. However, @lukasfrank wanted to have this sway. Let's finalize it offline.

Reply:
As per our offline discussion, this aligns with our current development standards, so we'll proceed with this approach.